content/200-orm/200-prisma-client/500-deployment/301-edge/450-deploy-to-cloudflare.mdx

```diff
@@ -0,0 +1,526 @@
+```
+
+#### 5. Use Prisma Client in your Worker to send a query to the database
+
+Here is a sample code snippet that you can code to instantiate `PrismaClient` and send a query to your database:
+
+```ts
+import { PrismaClient } from '@prisma/client'
+import { PrismaPg } from '@prisma/adapter-pg'
+import { Pool } from 'pg'
+
+export interface Env {
+  DATABASE_URL: string
+}
+
+export default {
+  async fetch(_: any, env: Env) {
+    const pool = new Pool({ connectionString: env.DATABASE_URL })
+    const adapter = new PrismaPg(pool)
+    const prisma = new PrismaClient({ adapter })
+
+    const users = await prisma.user.findMany()
+    const result = JSON.stringify(users)
+    return new Response(result)
+  },
+}
+```
+
+#### 6. Run the Worker locally
+
+To run the Worker locally, you can run the `wrangler dev` command:
+
+```terminal
+npx wrangler dev
+```
+
+#### 7. Set the `DATABASE_URL` environment variable and deploy the worker
+
+To deploy the worker, you first need to the `DATABASE_URL` environment variable [via the `wrangler` CLI](https://developers.cloudflare.com/workers/configuration/secrets/#secrets-on-deployed-workers):
+
+```terminal
+npx wrangler secret put DATABASE_URL
+```
+
+The command is interactive and will ask you to enter the value for the `DATABASE_URL` env var as the next step in the terminal.
+
+> **Note**: This command requires you to be authenticated, and will ask you to log in to your Cloudflare account in case you are not.
+
+Then you can go ahead then deploy the worker:
+
+```terminal
+npx wrangler deploy
+```
+
+### PlanetScale
+
+If you are using a PlanetScale database, you need to:
+
+- use the `@prisma/adapter-planetscale` database adapter (via the `driverAdapters` Preview feature)
+- manually remove the conflicting `cache`field ([learn more](<https://developers.cloudflare.com/workers/databases/native-integrations/planetscale/#:~:text=fetch%3A%20(,init)%3B>)):
+
+  ```ts
+  export default {
+    async fetch(request, env, ctx) {
+      const client = new Client({
+        url: env.DATABASE_URL,
+        // see https://github.com/cloudflare/workerd/issues/698
+        fetch(url, init) {
+          delete init['cache']
+          return fetch(url, init)
+        },
+      })
+      const adapter = new PrismaPlanetScale(client)
+      const prisma = new PrismaClient({ adapter })
+
+      // ...
+    },
+  }
+  ```
+
+#### 1. Configure Prisma schema & database connection
+
+First, ensure that database connection is configured properly. In your Prisma schema, set the `url` of the `datasource` block to the `DATABASE_URL` environment variable. You also need to enable the `driverAdapters` feature flag:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["driverAdapters"]
+}
+
+datasource db {
+  provider     = "mysql"
+  url          = env("DATABASE_URL")
+  relationMode = "prisma" // recommended for PlanetScale DBs
```

Comment:
```suggestion
  relationMode = "prisma" // required for PlanetScale, as by default, Foreign Keys are disabled 
```
See https://planetscale.com/docs/concepts/foreign-key-constraints

Reply:
You will not be able to migrate your database schema as it will try to create FKs that are forbidden by Planetscale in the default configuration.